java/client/src/test/java/glide/api/RedisClusterClientTest.java

```diff
@@ -0,0 +1,260 @@
+        assertEquals(value, payload);
+    }
+
+    @SneakyThrows
+    @Test
+    public void customCommand_interruptedException() {
+        // setup
+        String key = "testKey";
+        String cmd = "GETSTRING";
+        String[] arguments = new String[] {cmd, key};
+        CompletableFuture<ClusterValue<Object>> testResponse = mock(CompletableFuture.class);
+        InterruptedException interruptedException = new InterruptedException();
+        when(testResponse.get()).thenThrow(interruptedException);
+        when(commandManager.<ClusterValue<Object>>submitNewCommand(
+                        eq(CUSTOM_COMMAND), any(), any(), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        InterruptedException exception =
+                assertThrows(
+                        InterruptedException.class,
+                        () -> {
+                            CompletableFuture<ClusterValue<Object>> response = service.customCommand(arguments);
+                            response.get();
+                        });
+
+        // verify
+        assertEquals(interruptedException, exception);
+    }
+
+    @SneakyThrows
+    @Test
+    public void info_returns_string() {
+        // setup
+        CompletableFuture<ClusterValue<Map>> testResponse = mock(CompletableFuture.class);
+        Map testPayload = new HashMap<String, String>();
+        testPayload.put("addr1", "value1");
+        testPayload.put("addr1", "value2");
+        testPayload.put("addr1", "value3");
+        when(testResponse.get()).thenReturn(ClusterValue.of(testPayload));
+        when(commandManager.<ClusterValue<Map>>submitNewCommand(
+                        eq(INFO), eq(new String[0]), any(), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<ClusterValue<String>> response = service.info();
+
+        // verify
+        assertEquals(testResponse, response);
+        ClusterValue<String> clusterValue = response.get();
+        assertTrue(clusterValue.hasMultiData());
+        Map<String, String> payload = clusterValue.getMultiValue();
+        assertEquals(testPayload, payload);
+    }
+
+    @SneakyThrows
+    @Test
+    public void info_with_route_returns_string() {
+        // setup
+        CompletableFuture<ClusterValue<Map>> testResponse = mock(CompletableFuture.class);
+        Map<String, String> testClusterValue = Map.of("addr1", "addr1 result", "addr2", "addr2 result");
+        RequestRoutingConfiguration.Route route = RequestRoutingConfiguration.SimpleRoute.ALL_NODES;
+        when(testResponse.get()).thenReturn(ClusterValue.of(testClusterValue));
+        when(commandManager.<ClusterValue<Map>>submitNewCommand(eq(INFO), any(), any(), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<ClusterValue<String>> response = service.info(route);
+
+        // verify
+        assertEquals(testResponse, response);
```

Comment:
![image](https://github.com/Bit-Quill/glide-for-redis/assets/88679692/3d0b06b4-23a7-4593-a37d-1d26ed62fc8f)